catalog/views.py

```diff
@@ -30,11 +31,20 @@ class ProductDetailView(DetailView):
     model = Product
 
 
+
+
 class ProductCreateView(LoginRequiredMixin, CreateView):
     model = Product
     form_class = ProductForm
     success_url = reverse_lazy("products:products_list")
 
+    def form_valid(self, form):
+        product = form.save()
```

Comment:
На каждый save ты делаешь запрос в базу - сначала INSERT, потом UPDATE, выставляя владельца. Правильнее сделать это либо в одной транзакции, либо в первый раз не делать коммит
```suggestion
        product = form.save(commit=False)
```